kernel/drivers/LegacyATA.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <libsystem/thread/Lock.h>
+
+#include "kernel/devices/LegacyDevice.h"
+
+class LegacyATA : public LegacyDevice
+{
+private:
+    Lock _buffer_lock{"legacy-ata"};
```

Comment:
This lock is not needed Device has already the same lock ;)

Reply:
uhm, can't find any lock inside a base class

Reply:
My bad fsnode has one